### ISSUE TITLE
Lab 4 Level 2/3

### ISSUE THREAD

**Issue by Monoji77** [nus-cs2030/2324-s2#275]
I used ImmutableMap<String, String> where the key is (Level2: name of assessment; Level3: name of course) and the value is (Level2: Assessment.toString(); Level3: Course.toString()). 

Then under my course toString() and Student toString(), I used iterator() method in ImmutableMap and made a custom print since the toString() in ImmutableMap uses "=" instead of ":".  I was able to get the correct output but I was wondering if this was the intended implementation? It seems a bit wrong since my toString() method is lengthy and my ImmutableMap<String, String> uses Strings.

**Comment by jorized** [comment]
I think it should be the correct implementation as I have similar codes as you when doing level 1 and 2, iterating the immutableMap in the toString() method in both Course and Student class. Since it is also said in CodeCrunch that "You will notice that the implementations of the Student and Course classes are very similar." in level 3, I think it seems about right.

**Comment by xiang-qiuyu** [comment]
i think it's fine! i had to loop every entry in my ImmutableMap also just to sieve out the values and concatenate them one by one. as long as you can pass level 1 and level 2, your toString() method should be fine.

**Comment by TongZhengHong** [comment]
Yes, that is correct, you have to override the toString method to get the desired `:` in the output. 
Moving on to level 3, you will only need to override the toString method in `KeyableMap` since `Student` and `Course` will most likely inherit `KeyableMap`

**Comment by Monoji77** [comment]
Okay i got it thanks guys :)
